src/matcalc/_md.py

```diff
@@ -0,0 +1,322 @@
+        as required by the Nose-Hoover NPT implementation.
+
+        Parameters:
+            atoms (Atoms): ASE atoms object.
+        """
+        if not atoms.cell[1, 0] == atoms.cell[2, 0] == atoms.cell[2, 1] == 0.0:
+            a, b, c, alpha, beta, gamma = atoms.cell.cellpar()
+            angles = np.radians((alpha, beta, gamma))
+            sin_a, sin_b, _ = np.sin(angles)
+            cos_a, cos_b, cos_g = np.cos(angles)
+            cos_p = (cos_g - cos_a * cos_b) / (sin_a * sin_b)
+            cos_p = np.clip(cos_p, -1, 1)
+            sin_p = np.sqrt(1 - cos_p**2)
+            new_basis = [
+                (a * sin_b * sin_p, a * sin_b * cos_p, a * cos_b),
+                (0, b * sin_a, b * cos_a),
+                (0, 0, c),
+            ]
+            atoms.set_cell(new_basis, scale_atoms=True)
+
+    def calc(self, structure: Structure | dict[str, Any]) -> dict:
+        """
+        Performs the MD simulation calculation for the input structure. Prior to generating initial velocities,
+        this method calls the superclass's calc method for preprocessing and optionally relaxes the structure.
+        It then initializes the atomic velocities using the Maxwell-Boltzmann distribution, runs the MD simulation,
+        and returns the final energy along with the final atomic configuration.
+
+        Parameters:
+            structure (Structure | dict[str, Any]): Input structure as a Structure instance or a dictionary.
+
+        Returns:
+            dict: A dictionary containing the final energy and the final atomic configuration.
+                  It includes the keys "final_structure" and "energy".
+        """
+        # Preprocess the input structure using the superclass's calc method
+        result = super().calc(structure)
+        structure_in: Structure = result["final_structure"]
+
+        # If relaxation is enabled, relax the structure before MD simulation
+        if self.relax_structure:
+            relaxer = RelaxCalc(
+                self.calculator,
+                fmax=self.fmax,
+                optimizer=self.optimizer,
+                **(self.relax_calc_kwargs or {}),
+            )
+            result |= relaxer.calc(structure_in)
+            structure_in = result["final_structure"]
+
+        # Convert the relaxed structure to an ASE atoms object
+        atoms = AseAtomsAdaptor().get_atoms(structure_in)
+        # Initialize atomic velocities using the Maxwell-Boltzmann distribution
+        MaxwellBoltzmannDistribution(atoms, temperature_K=self.temperature)
+        # Initialize and run the MD simulation
+        md = self._initialize_md(atoms)
+        md.run(self.steps)
+        # Update the result dictionary with the final frame and energy
+        structure_out = AseAtomsAdaptor().get_structure(atoms)
+
+        energy_pot = atoms.get_potential_energy()
+        energy_kin = atoms.get_kinetic_energy()
+        energy_tot = atoms.get_total_energy()
+
+        result |= {
```

Comment:
There needs to be an option to export the entire trajectory with all energies. It is rare you just need the final structure and energy. If you have the entire trajectory, the DiffusionCalc can simply call MDCalc to get the result and then perform an additional analysis.